framework/src/executioners/Transient.C

```diff
@@ -276,6 +276,11 @@ Transient::init()
     //  _dt = computeConstrainedDT();
     _dt = getDT();
   }
+  if (_dt == 0)
```

Comment:
I am thinking negative `_dt` could be allowed, for example, marching backward for adjoint calculation. So only checked `_dt == 0` instead of `_dt <= 0` here.